elastic/beats
# System test for beat module, xpack code path

Similar to https://github.com/elastic/beats/pull/12506 but for the `beat` module.

This PR introduces a system test to exercise the xpack code path (`xpack.enabled: true`) of the Beat Metricbeat module. 

### Testing this PR

1. Build and activate the virtual env for system tests.
   ```
   cd metricbeat
   make python-env
   . build/python-env/bin/activate
   ```

1. Make sure `fields.yml` is present.
   ```
   make update
   ```

1. Build the test binary.
   ```
   make metricbeat.test
   ```

1. Run the integration tests

   ```
   INTEGRATION_TESTS=1  nosetests -v --nocapture tests/system/test_beat.py
   ```

1. Assert that there are no errors.
